tensorboard/webapp/plugins/npmi/views/data_selection/metric_search/metric_search_component.ts

```diff
@@ -0,0 +1,63 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  HostBinding,
+  ViewChild,
+} from '@angular/core';
+import {MatAutocompleteTrigger} from '@angular/material/autocomplete';
+
+import {Store} from '@ngrx/store';
+
+import * as npmiActions from '../../../actions';
+
+@Component({
+  selector: 'metric-search-component',
+  templateUrl: './metric_search_component.ng.html',
+  styleUrls: ['./metric_search_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetricSearchComponent {
+  @Input() completions!: string[];
+  @Input() regexFilterValue!: string;
+  @HostBinding('class.valid') @Input() isRegexFilterValid!: boolean;
+  @Output() onRegexFilterValueChange = new EventEmitter<string>();
+  @ViewChild(MatAutocompleteTrigger, {static: true})
+  autocompleteTrigger!: MatAutocompleteTrigger;
+
+  constructor(private store: Store<any>) {}
```

Comment:
Sorry, we don't currently have great documentation on ngrx patterns.  However, our pattern so far is to make `component`s simple and never depend on the store.  By keeping them pure and having `container`s do all the store-interaction, the hope is that it keeps things simpler to reason about.

Can we remove this Store injection, and replace the `this.store.dispatch` calls on L60-61 with something that notifies metric_search_container that it needs to do dispatches?

Reply:
Ahh did not know this was implemented this way for actions as well. Changed it.